src/game/objects/bridge.c

```diff
@@ -137,27 +137,38 @@ void BridgeFlatCeiling(
     }
 }
 
-int32_t GetOffset(ITEM_INFO *item, int32_t x, int32_t z)
+int32_t GetOffset(ITEM_INFO *item, int32_t x, int32_t y, int32_t z)
 {
-    if (item->pos.y_rot == 0) {
-        return (WALL_L - x) & (WALL_L - 1);
-    } else if (item->pos.y_rot == -PHD_180) {
-        return x & (WALL_L - 1);
-    } else if (item->pos.y_rot == PHD_90) {
-        return z & (WALL_L - 1);
-    } else {
-        return (WALL_L - z) & (WALL_L - 1);
+    int32_t offset = 0;
+    if (item->pos.y_rot == 0)
+        offset = -x & 0x3FF;
+    else if (item->pos.y_rot == -0x8000)
+        offset = x & 0x3FF;
+    else if (item->pos.y_rot == 0x4000)
+        offset = z & 0x3FF;
+    else {
+        offset = -z & 0x3FF;
+        // Fixes edge case of an invisible wall on the tilt2 bridge floor
+        // Offset would get set to 0 on a specific z pos on bottom of slope
+        // This sets the offset to the max value (1023) when Lara's at the
+        // bottom of the slope
+        if (g_Config.fix_bridge_grabs && offset == 0) {
+            if (y < item->pos.y) {
+                offset = (-z - 1) & 0x3FF;
+            }
+        }
```

Comment:
use the constants not hard coded values ;)

Reply:
-x and WALL_L-x are not the same, does this not cause more issues? as it seems from your comments the only issue was in the last case?

Reply:
Changed back to TR1 version of this.